document-store/src/main/java/org/hypertrace/core/documentstore/model/config/TypesafeConfigDatastoreConfigExtractor.java

```diff
@@ -49,16 +53,41 @@ public static TypesafeConfigDatastoreConfigExtractor from(
     return new TypesafeConfigDatastoreConfigExtractor(config, type);
   }
 
+  public TypesafeConfigDatastoreConfigExtractor keysForEndpoints(
+      @NonNull final String endpointsBaseKey,
+      @NonNull final String hostKey,
+      @NonNull final String portKey) {
+    if (!config.hasPath(endpointsBaseKey)) {
+      return this;
+    }
+
+    for (final Config endpointConfig : config.getConfigList(endpointsBaseKey)) {
+      final Endpoint.EndpointBuilder builder = Endpoint.builder();
+
+      if (endpointConfig.hasPath(hostKey)) {
+        builder.host(endpointConfig.getString(hostKey));
+      }
+
+      if (endpointConfig.hasPath(portKey)) {
```

Comment:
If its missing, will it use default port?

Reply:
Yes. Mongo and Postgres constructors make sure to fill-in the respective default ports if missing.